tests/unit/decorateWithMetrics.js

```diff
@@ -0,0 +1,235 @@
+import {
+    decorateWithMetrics,
+    getMetrics,
+    resetMetrics
+} from '../../lib/decorateWithMetrics';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+
+describe('decorateWithMetrics', () => {
+    beforeEach(() => {
+        resetMetrics();
+    });
+
+    it('Should collect metrics for a single method, single call', () => {
```

Comment:
For unit tests, we like to have them contain comments in the following format:

```
// GIVEN <some initial condition>
// WHEN <an action happens>
// THEN <a result is expected>
```

I know some of the tests in this repo aren't very consistent with that, but we have found that providing comments in this style is extremely helpful to convey context and improve maintainability.

Reply:
I think I can apply it like this:

// GIVEN a mock function that returns a mock object
// WHEN it is decorated and executed
// THEN calling `getMetrics` should return data about the particular call

Or 

// GIVEN a mock function, returning a mock object that is decorated
// WHEN it is called with a parameter
// THEN  calling `getMetrics` should return data about the particular call